### PR TITLE
refactor: clean up types some more

### LINKED ISSUE #1465: [Zoe/ERTP] Typing fixes from Mark's Zoe Release Candidate Review
For context, these are typing-related review comments (yes, each link is different):

- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469542420
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r468144498
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r468146840
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r468978796
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r468986037
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r468987573
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r468990391
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r468994223
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r468994784
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469574981
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469582691
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469582749
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469591773
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469597713
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469605060
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469615154
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469689542
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469689884
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469693275
- [x] https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r469694785

### LINKED ISSUE #1707: zcf.makeZCFMint types are wrong. Needs to return Promise<ZCFMint>
**What is the Problem Being Solved?**

`makeZCFMint` should return a promise for a ZCFMint.

We should change the types to:
```
/**
 * @callback MakeZCFMint
 * @param {Keyword} keyword
 * @param {AmountMathKind=} amountMathKind
 * @returns {Promise<ZCFMint>}
 */
```

### PR DESCRIPTION
**Pull request by michaelfig**
Closes #1707, closes #1465

### CODE REVIEW

**Review by erights — APPROVED** [review]
LGTM

File: packages/cosmic-swingset/lib/ag-solo/vats/lib-board.js [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
 import { generateSparseInts } from '@agoric/sparse-ints';
 import { assert, details } from '@agoric/assert';
 import makeStore from '@agoric/store';
 
+/**
+ * @typedef {Object} Board
+ * @property {(id: string) => any} getValue
+ * @property {(value: any) => string} getId
+ * @property {(value: any) => boolean} has
+ * @property {() => string[]} ids```
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/cosmic-swingset/lib/ag-solo/vats/lib-board.js:13]: I don't know.  @katelynsills ?

File: packages/zoe/docs/seats.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ The type of the ZCFSeat is:
 /**
  * @typedef {Object} ZCFSeat
  * @property {() => void} exit
- * @property {(msg: string=) => never} kickOut
+ * @property {(msg?: string) => never} kickOut
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/zoe/docs/seats.md:59]: This is in a documentation file.  The real type is correct.

File: packages/cosmic-swingset/lib/ag-solo/vats/lib-board.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/cosmic-swingset/lib/ag-solo/vats/lib-board.js:13]: I believe it was a design by Dean to show to users that the Ids are not meant to be secret.

**Comment by katelynsills** [comment]
Thanks so much!

**Pull request merged**